Update database guides with database configurator. (#10451)
gravitational/teleport: docs/pages/includes/database-access/aws-bootstrap.mdx

Teleport can bootstrap IAM permissions for the Database Service based on its

> configuration. You can use bootstrap in automatic or manual mode:

Reviewer: "You can use bootstrap" suggests that "bootstrap" is the name for this feature. Is it? Should we capitalize it? Otherwise, to be clearer on the feature name, I think we can say something like,

```markdown
Teleport can bootstrap IAM permissions for the Database Service based on its
configuration using the `teleport db configure bootstrap` command. You can use
this command in automatic or manual mode:
```

Author: I think the more appropriate name would be "database configurator", but I like the idea of placing the command instead of the feature name.

- In automatic mode, Teleport will attempt to create appropriate IAM policies
  and attach them to the specified IAM identity (user or role). This requires
  IAM permissions to create and attach IAM policies.
- In manual mode, Teleport will print required IAM policies. You can then create
  and attach them manually using AWS management console.

<Admonition type="note" title="AWS credentials">
  AWS Credentials are only required if you’re running the command in "automatic"
  mode. The command uses the default credential provider chain to find AWS
  credentials. See [Specifying Credentials](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials) for more information.
</Admonition>

<Tabs>
  <TabItem label="Automatic / IAM User">
  Use this command to bootstrap the permissions automatically when

>   your Teleport database agent runs as an IAM user (for example, uses AWS

Reviewer: ```suggestion
  your Teleport database agent runs as an IAM user (for example, uses an AWS
```

  credentials file).

  ```code
  $ teleport db configure bootstrap -c /etc/teleport.yaml --attach-to-user TeleportUser
  ```
  </TabItem>

  <TabItem label="Automatic / IAM Role">
  Use this command to bootstrap the permissions automatically when
  your Teleport database agent runs as an IAM role (for example, on an EC2

>   instance with attached IAM role).

Reviewer: ```suggestion
  instance with an attached IAM role).
```

  ```code
  $ teleport db configure bootstrap -c /etc/teleport.yaml --attach-to-role TeleportRole
  ```
  </TabItem>

  <TabItem label="Manual">
  Use this command to display required IAM policies which you will then create in your AWS console:

  ```code
  $ teleport db configure bootstrap -c /etc/teleport.yaml --manual
  ```
  </TabItem>
</Tabs>

> See full bootstrap command

Reviewer: ```suggestion
See the full `bootstrap` command
```

Teleport can bootstrap IAM permissions for the Database Service based on its
configuration using the `teleport db configure bootstrap` command. You can use
this command in automatic or manual mode:
- In automatic mode, Teleport will attempt to create appropriate IAM policies
  and attach them to the specified IAM identity (user or role). This requires
  IAM permissions to create and attach IAM policies.
- In manual mode, Teleport will print required IAM policies. You can then create

>   and attach them manually using AWS management console.

Reviewer: ```suggestion
  and attach them manually using the AWS management console.
```

> The exact set of required permissions depends on whether you're connecting to

Reviewer 2: I would remove this first sentence.

different AWS-hosted databases, as well as the IAM identity your Teleport
database agent will be using (user or role). The database bootstrap command
generates custom AWS policies based on your configuration. Bootstrap can be

> performed in two modes:

Reviewer 2: I would rephrase this a bit to something like:

```
Teleport can bootstrap IAM permissions for the Database Service based on its configuration.
You can use bootstrap in automatic or manual mode:
```

> - Automatic: Generates and creates IAM policy and policy boundary and attaches
>   to provided target (user or role). In this command, the command requires AWS
>   credentials set to identity with enough permissions;

Reviewer 2: ```suggestion
- In automatic mode, Teleport will attempt to create appropriate IAM policies and attach
  them to the specified IAM identity (user or role). This requires IAM permissions to create
  and attach IAM policies.
```

> - Manual: Generates IAM policy and boundary and prints to STDOUT. You then can
>   add them to your AWS account manually (for instance, using the AWS management
>   console);

Reviewer 2: ```suggestion
- In manual mode, Teleport will print required IAM policies. You can then
  create and attach them manually using AWS management console.
```

<Admonition type="note" title="AWS credentials">
  AWS Credentials are only required if you’re running the command in "automatic"
  mode. The command uses the default credential provider chain to find AWS
  credentials. See [Specifying Credentials](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials) for more information.
</Admonition>

<Tabs>
  <TabItem label="Automatic / IAM User">

>   Use this command if you’re bootstrapping the permissions automatically and

Reviewer 2: ```suggestion
  Use this command to bootstrap the permissions automatically when
```

  your Teleport database agent runs as an IAM user (for example, uses AWS
  credentials file).

  ```code
  $ teleport db configure bootstrap -c /etc/teleport.yaml --attach-to-user TeleportUser
  ```
  </TabItem>

  <TabItem label="Automatic / IAM Role">

>   Use this command if you’re bootstrapping the permissions automatically and

Reviewer 2: ```suggestion
  Use this command to bootstrap the permissions automatically when
```

  your Teleport database agent runs as an IAM role (for example, on an EC2
  instance with attached IAM role).

  ```code
  $ teleport db configure bootstrap -c /etc/teleport.yaml --attach-to-role TeleportRole
  ```
  </TabItem>

  <TabItem label="Manual">

>   Use this command if you’re bootstrapping the permissions manually, and attach
>   the generated permissions to your target (user or role).

Reviewer 2: ```suggestion
  Use this command to display required IAM policies which you will then create in your AWS console:
```

  ```code
  $ teleport db configure bootstrap -c /etc/teleport.yaml --manual
  ```
  </TabItem>
</Tabs>

<Admonition type="tip" title="Tip">
  The bootstrap command support other customizations, such as changing the IAM
  policy name (`--policy-name`). For more options, check out the [command
  reference page.](../../database-access/reference/cli.mdx#teleport-db-configure-bootstrap)

> </Admonition>

Reviewer 2: Instead of admonition, I would just say `See full bootstrap command [reference](link-here)`.
